modules/roles_profiles/manifests/profiles/scriptworker_user.pp

```diff
@@ -0,0 +1,24 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+class roles_profiles::profiles::scriptworker_user {
+    case $::hostname {
+        /^dep-mac-v3-signing\d+/: {
+            class { 'signing_worker::system_user':
```

Comment:
It's worked on the test nodes (and I lifted the user creation from the existing code). You mentioned there were some oddities around creating system users that you'd implemented workarounds for, would that be covered if I used resources?

Reply:
Fact is a classes should only be declared once if using resource-like declarations.  eg `class { 'signing_worker::system_user':}` because you aren't instantiating multiple instances of the class, only overriding a single instance with each declaration of that class at compile time.  Depending on puppets order of execution the results are unreliable.  And this isn't the behavior you want.  You want to instantiate these code blocks as multiple objects with different user parameters in the same catalog at the same time, therefore, you should user the defined resource block instead.  A defined resource block can instantiate multiple times because each has its own unique separate namespace.

See: [https://puppet.com/docs/puppet/latest/lang_classes.html#section-x54-1hk-xhb](https://puppet.com/docs/puppet/latest/lang_classes.html#section-x54-1hk-xhb)

Also, if this is being done elsewhere in the code, let me know so I can fix it.  I went looking but couldn't find it.